### PR TITLE
Switch from black+flake8 to ruff

### PR DESCRIPTION
**Pull request by samuelhwilliams**
### Change description
Apply ruff formatting changes across the codebase, prior to lift+shift.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -37,26 +33,45 @@ max_attribute_length=1000   # ^^^
...
+    "C90",  # mccabe cyclomatic complexity
+    "G",  # flake8-logging-format
+]
+ignore = []
+exclude = [
+    "db/migrations/versions/",
+    "venv*",
+    ".venv*",
+    "__pycache__",
+    "fund_store/config/fund_loader_config/FAB/"
```
Comment (MarcUsher) [inline pyproject.toml:57]: We prob don't need the `db/migrations/versions/` and `fund_store` exceptions here, unless we're just pre-empting the eventual merge?

File: app/default/data.py [hunk 2]
```diff
@@ -210,7 +210,7 @@ def get_application_display_config(fund_id, round_id, language):
     try:
         return [ApplicationMapping.from_dict(section) for section in application_display_response]
     except Exception as e:
-        raise ValueError(f"Failed to create ApplicationMapping instance: {str(e)}")
+        raise ValueError("Failed to create ApplicationMapping instance") from e
```
Comment (MarcUsher) [inline app/default/data.py:213]: Do we want to include the error or some of its info in the ValueError being raised?

**Review by MarcUsher — APPROVED** [review]
Looks good! Approving but have just left 2 tiny 🤏  comments.

File: app/default/data.py (hunk 2 above)
Comment (samuelhwilliams) [reply app/default/data.py:213]: The info from the previous error gets chained as context into this new error via the `from e`, so the context should be visible from the stacktrace still.

File: pyproject.toml (hunk 1 above)
Comment (samuelhwilliams) [reply pyproject.toml:57]: Just a lazy copy paste on my side. I'm gonna continue to be lazy and leave them, since we can rationalise in the combination...

**Pull request merged**